spec/site/alarm_spec.rb

```diff
@@ -67,4 +67,43 @@ def check_scripts
   #    system(SCRIPT_PATHS['deactivate_alarm'])
   #  end
   #end
+
+  it 'Disruption test', :script, sxl: '>=1.0.7' do |example|
+    check_scripts
+    component = COMPONENT_CONFIG['detector_logic'].keys.first
+
+    TestSite.isolated do |task,supervisor,site|
```

Comment:
you should use TestSite.disconnected, and and place the system() call inside that block.
TestSite.isolated will first connect, when disconnect, which will add unnecessary delay to the test

Reply:
the ekstra delay will occur in case that site is already disconnected. in that case there is no need to connect and then disconnect. using TestSite.disconnected will see that it's already offline and just go ahead.

Reply:
Right, I'll change it. I take it that no explicit connection check is needed as part of the test then?

Reply:
TestSite.disconnected doesn't work unfortunately. See issue #67